File: solutions/typescript/sum-of-multiples/1/sum-of-multiples.ts
export function sum(multiples: number[], level: number): number {
  if (level <= 0) return 0;
  
  // Remove zeros and negatives, and deduplicate
  const bases = [...new Set(multiples.filter(m => m > 0))];
  
  if (bases.length === 0) return 0;
  
  let total = 0;
  
  for (let i = 1; i < level; i++) {
    if (bases.some(base => i % base === 0)) {
      total += i;
    }
  }
  
  return total;
}

File: solutions/java/sum-of-multiples/1/src/main/java/SumOfMultiples.java
import java.util.HashSet;
import java.util.Set;

class SumOfMultiples {
    private final int level;
    private final int[] bases;
    
    SumOfMultiples(int number, int[] set) {
        this.level = number;
        this.bases = set;
    }

    int getSum() {
        Set<Integer> multiples = new HashSet<>();
        
        for (int base : bases) {
            // Skip invalid bases (0 or negative) to avoid infinite loops
            if (base <= 0) {
                continue;
            }
            
            // Find all multiples of base that are less than level
            for (int multiple = base; multiple < level; multiple += base) {
                multiples.add(multiple);
            }
        }
        
        // Sum all unique multiples
        int sum = 0;
        for (int m : multiples) {
            sum += m;
        }
        return sum;
    }
}